src/components/FeatureViews/Courses/ModalTextEditor.js

```diff
@@ -134,122 +185,309 @@ const ModalTextEditor = ({
+    },
+    STUDENT_ENROLLMENT: {
+      subject: subject,
+      body: body.getCurrentContent(),
+      userId,
+      posterId: poster_id,
+    },
+    COURSE_SESSIONS: {
+      subject: subject,
+      body: convertToRaw(body.getCurrentContent()),
+      user: poster_id,
+      sessionId: sessionId,
+      id: noteId,
+    },
+  };
+
+  const [mutateTextEditor, createResults] = useMutation(MUTATIONS[origin], {
+    onCompleted: () => handleClose(false),
+    update: (cache, { data }) => {
+      const [newTextData] = Object.values(data[MUTATION_KEY[origin]]);
+      const cachedTextData = cache.readQuery({
+        query: QUERIES[origin],
+        variables: QUERY_VARAIBLES[origin],
+      })[QUERY_KEY[origin]];
+      let updatedTextData = [...cachedTextData];
+      const matchingIndex = updatedTextData.findIndex(
+        ({ id }) => id === newTextData.id
+      );
+      if (matchingIndex === -1) {
+        updatedTextData = [...cachedTextData, newTextData];
+      } else {
+        updatedTextData[matchingIndex] = newTextData;
+      }
+
+      cache.writeQuery({
+        data: {
+          [QUERY_KEY[origin]]: updatedTextData,
         },
+        query: QUERIES[origin],
+        variables: QUERY_VARAIBLES[origin],
       });
+    },
+  });
+
+  const handleClose = () => {
+    handleCloseForm(false);
+    setBody(EditorState.createEmpty());
+  };
+
+  const handleSubmit = async e => {
+    e.preventDefault();
+    const createTextData = await mutateTextEditor({
+      variables: MUTATION_VARIABLES[origin],
+    });
+    setBody(EditorState.createEmpty())
+  };
+
+  const toggleInlineStyles = e => {
+    e.preventDefault();
+    const style = e.currentTarget.getAttribute("data-style");
+    const { toggleInlineStyle, toggleBlockType } = RichUtils
+    const newState = {
+      [style]: toggleInlineStyle(body, style),
+      "unordered-list-item": toggleBlockType(body, style),
+      "ordered-list-item": toggleBlockType(body, style),
+    }; 
+    if(newState[style]) {
+      setBody(newState[style])
+      return "handled"
+    }
+    return "not handled"
+  };
+
+  const handleKeyCommand = (command, body) => {
+    const newState = RichUtils.handleKeyCommand(body, command);
+    if (newState) {
+      setBody(newState);
+      return "handled";
     }
+    return "not handled";
   };
 
-  const handleSubjectChange = useCallback((event) => {
+  const handleCheckboxChange = setCheckbox => e =>
+    setCheckbox(e.target.checked);
+
+  const handleSubjectChange = useCallback(event => {
     setSubject(event.target.value);
   }, []);
 
-  const handleBodyChange = useCallback((event) => {
-    setBody(event.target.value);
+  const handleBodyChange = useCallback(event => {
+    setBody(event);
   }, []);
 
+  const renderButtonText = () => {
+    switch (buttonState) {
+      case "edit":
+        return "EDIT";
+      case "post":
+        return "POST";
+      default:
+        return "ADD NOTE";
+    }
+  };
```

Comment:
Optional: since you only want static values, a lookup table might be more appropriate, e.g.

    const buttonTextParser = {
        "edit": EDIT,
        ...

using a default value if the key does not exist. This may or may not be more complex, depending on the code structure; check for yourself if it's worth converting.

Also, since you only plan to compute this once whenever buttonState is a certain value, a computed value would be more appropriate; call the function, store the value into a constant (perhaps via `useMemo`) and then used that constant, for both simplicity and a bit of optimization.